### PR TITLE
send begin/end notifications when installing types packages

### PR DESCRIPTION
**Pull request by vladima**
fixes https://github.com/Microsoft/TypeScript/issues/12540

// cc @mhegazy 

open question: do we assume that existing consumers of tsserver will ignore new event or should we allow new events under the command line flag

### CODE REVIEW

File: src/server/protocol.ts [hunk 1]
```diff
@@ -2117,6 +2117,40 @@ namespace ts.server.protocol {
...
+        body: BeginInstallTypesEventBody;
+    }
+
+    export interface EndInstallTypesEvent extends Event {
+        event: EndInstallTypesEventName;
+        body: EndInstallTypesEventBody;
+    }
+
+    export interface InstallTypesEventBody {
+        /**
```
Comment (mhegazy) [inline src/server/protocol.ts]: missing `typingsInstallerVersion`.

File: src/server/server.ts [hunk 2]
```diff
@@ -291,12 +291,30 @@ namespace ts.server {
...
+                this.eventSender.event(body, eventName);
+
+                return;
+            }
+
+            if (response.kind === EventEndInstallTypes) {
+                if (!this.eventSender) {
+                    return;
+                }
+                if (this.telemetryEnabled) {
```
Comment (mhegazy) [inline src/server/server.ts]: why do we still need `TypingsInstalledTelemetryEvent*`?

Comment (vladima) [inline src/server/server.ts]: reasoning behind this was: these events are semantically different. one is intended to be consumed only in the editor, another - send by VSCode on our behalf. In theory the latter one can be sent by just grabbing the payload and sending it as is treating its content as opaque. Yes, in this case VSCode can reconstruct telemetry event from the content of `End` notification, however I'd prefer to keep them separate for simplify consumption. // cc @mjbvz for his opinion

Comment (mjbvz) [inline src/server/server.ts]: Yes, I think that keeping both makes sense since they are used for different purposes.

**Pull request merged**